midp/content.js

```diff
@@ -4,10 +4,12 @@
 'use strict';
 
 var Content = (function() {
-  var chRegisteredClassName = null;
-  var chRegisteredStorageID = -1;
-  var chRegisteredID = null;
-  var chRegisteredRegistrationMethod = -1;
+  // Allow configuring these values for MIDlets that don't register
+  // the content handler in time.
```

Comment:
I'm a little leery of implementing dynamic registration in terms of static configuration that will have to be set for each MIDlet suite that uses this mechanism. What happens if a new version of a MIDlet changes one of these values? Will its configuration need to be updated and a new FirefoxOS app version pushed? If so, won't that version not work with the older MIDlet?

It seems like what we need is for the FG MIDlet never to launch until the BG MIDlet has finished launching. Somewhere you mentioned that this caused a startup perf regression. Do you have info on the method you used and how much it hurt startup time?